doc/adr/007_Typescript.md

```diff
@@ -0,0 +1,51 @@
+# Using Typescript
+
+Date: 2019-02-07
+
+## Status
+
+Accepted
+
+## Context
+
+During the modernization of the client-side code outlined in "[ADR 006 - Using Vue.js](006_Vue.js.md)" and the first prototype for that project - the Address Change Form - we also tried out using [TypeScript](https://www.typescriptlang.org/). After the project we evaluated if we want to continue to use TypeScript for client side code. We collected following arguments. 
+
+### Learning Curve
+Nobody of us has experience with TypeScript, there will be a learning curve and possible slowdowns while setting up features.
+
+### Migration Pain
+Migrating the existing JavaScript code looks daunting, we don't know if TypeScript will make the migration harder by having type requirements. However, since we're already migrating the code to Vue and Vuex, adding TypesScript into the mix will probably one be a minor hurdle.  
```

Comment:
My opinion is the opposite. We are going to move markup and logic to Vue but also move all store logic to Vuex. That's already quite the commitment, given we have many `.js` and `.twig` files.  Adding TypeScript to the loop adds more complexity to the task and I wouldn't consider it "minor hurdle". But since we already established this is going to be a rewrite and not refactoring we should come out of it with something fully done, which won't need any more iterations of improvements in the near future. Hence we do all the things (vue, vuex, typescript).

^ That doesn't sounds very ADRy :smile:  I'm sure you can formulate it better in case you agree to change this part.

Reply:
I've changed the wording. Is it what you wanted to say?

Reply:
yep, thanks!